src/manager/node.c

```diff
@@ -480,14 +490,16 @@ static int node_match_job_done(UNUSED sd_bus_message *m, UNUSED void *userdata,
         return 1;
 }
 
-static int node_match_heartbeat(UNUSED sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *error) {
-        char *node_name = NULL;
+static int node_match_heartbeat(UNUSED sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
+        Node *node = userdata;
 
-        int r = sd_bus_message_read(m, "s", &node_name);
+        struct timespec now;
+        int r = clock_gettime(CLOCK_REALTIME, &now);
         if (r < 0) {
-                hirte_log_errorf("Error reading heartbeat: %s", strerror(-r));
+                hirte_log_errorf("Failed to get current time on heartbeat: %s", strerror(-r));
                 return 0;
         }
+        node->last_seen = now.tv_sec;
```

Comment:
Maybe extract that to `time-util`? `get_time_seconds()` or something?

Reply:
Good point! I added another commit (so its easier to review), I'll rebase it into the other commits before merging. Could you have another look? @mkemel